test/brain/testing_forecast_util.cpp

```diff
@@ -110,5 +114,292 @@ matrix_eig TestingForecastUtil::GetWorkload(WorkloadType w, size_t num_samples,
   return data;
 }
 
+
+void TestingAugmentedNNUtil::Test(
+    brain::AugmentedNN &model, DistributionType d, 
+    size_t val_interval, size_t num_samples, 
+    float val_split, bool normalize, float val_loss_thresh, 
+    size_t early_stop_patience, float early_stop_delta) {
+  LOG_INFO("Using Model: %s", model.ToString().c_str());
+  size_t num_tests = model.GetBatchsize();
+  matrix_eig all_data = GetData(d, num_samples, num_tests);
+
+  matrix_eig test_data = all_data.bottomRows(num_tests*3);
+  matrix_eig data = all_data.topRows(all_data.rows() - num_tests*3);
+
+  brain::Normalizer n(normalize);
+  val_interval = std::min<size_t>(val_interval, model.GetEpochs());
+
+  // Determine the split point
+  size_t split_point =
+      data.rows() - static_cast<size_t>(data.rows() * val_split);
+
+  // Split into train/test data
```

Comment:
This is splitting into train/**validate** data, and the test data is separate, right?

Reply:
Right. I'll update the comment here.